Style guide + visual redesign

jrubenoff opened the pull request:
Alright, here's a start.

Right now, the `dvl-core` app mostly serves as a visual unit test and outdated pattern library. This PR:

- Contains a new, consistent visual design
- Adds a design spec and guidelines for developers
- Organizes _every_ existing component and brings the markup up-to-date
- Moves many styles for common components into here from Screendoor

![screen shot 2015-06-29 at 6 51 25 pm](https://cloud.githubusercontent.com/assets/1328849/8422415/e00f153e-1e8f-11e5-863e-db00cab7d35a.png)

Our apps will need some adjustment before merging this in, but this PR is intended to have as little external impact as possible. I'll be renaming components, changing others, and generally saving other "breaking changes" for subsequent releases.

This is mostly about how things look, so anyone on @dobtco/team is welcome to give feedback. Just follow these steps:

1. Clone this repo
2. Open Terminal.app, navigate to this folder
3. Type `script/bootstrap`, then `script/server`
4. Visit http://dvlcore.dev in your browser

dobthound commented on lib/dvl/simple_form_config.rb:
```diff
@@ -48,7 +48,7 @@
   config.item_wrapper_tag = false
 
-  config.label_text = lambda { |label, required, _| "#{label} #{required}" }
+  config.label_text = lambda { |label, required, _| "#{label}&nbsp;#{required}".html_safe }
```
Line is too long. [91/80]

ajb replied:
I'm curious: this is to ensure that the text doesn't break in between a label and the `abbr`?

jrubenoff replied:
Yup

dobthound commented on vendor/assets/javascripts/dvl/plugins/modernizr_custom.js:
```diff
@@ -0,0 +1,4 @@
+/* Modernizr 2.8.3 (Custom Build) | MIT & BSD
+ * Build: http://modernizr.com/download/#-boxshadow-cssgradients-shiv-cssclasses-testprop-testallprops-prefixes-domprefixes-load
```
Line is too long.

dobthound commented on vendor/assets/javascripts/dvl/plugins/modernizr_custom.js:
```diff
@@ -0,0 +1,4 @@
+/* Modernizr 2.8.3 (Custom Build) | MIT & BSD
+ * Build: http://modernizr.com/download/#-boxshadow-cssgradients-shiv-cssclasses-testprop-testallprops-prefixes-domprefixes-load
+ */
+;window.Modernizr=function(a,b,c){function y(a){j.cssText=a}function z(a,b){return y(m.join(a+";")+(b||""))}function A(a,b){return typeof a===b}function B(a,b){return!!~(""+a).indexOf(b)}function C(a,b){for(var d in a){var e=a[d];if(!B(e,"-")&&j[e]!==c)return b=="pfx"?e:!0}return!1}function D(a,b,d){for(var e in a){var f=b[a[e]];if(f!==c)return d===!1?a[e]:A(f,"function")?f.bind(d||b):f}return!1} ...
```
Line is too long.<br>Unnecessary semicolon.<br>Missing semicolon.<br>Unexpected '~'.<br>Expected '{' and instead saw 'return'.<br>The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.<br>Confusing use of '!'.<br>Missing 'new' prefix when invoking a constructor.<br>Expected '{' and instead saw 'throw'.<br>Missing '()' invoking a constructor.<br>A constructor name should start with an uppercase letter.<br>Expected an assignment or function call and instead saw an expression.<br>Expected '{' and instead saw 'x'.<br>Expected '{' and instead saw 'for'.<br>Too many errors. (100% scanned).

ajb commented:
:clap: 

Great job! This is truly a masterpiece... I really like how you've refined the visual styles.  All of my feedback is minor:

### Feedback on the content of the styleguide

- [ ] I'm not so sure about orange as a primary button color. Am I the only one that feels this way?
- [ ] Dropdown_select hover state http://take.ms/OnNYd
- [ ] There's a perceptible delay when checking radios and checkboxes (http://take.ms/bAvAt)
- [ ] Native <select> isn't filling 100% height (http://take.ms/Vb2Xp and http://take.ms/9OEKy)
- [ ] Input group text padding http://take.ms/3WCCD
- [ ] Subtle class? http://take.ms/6E4SI
- [ ] Small vs mini http://take.ms/IimZL
- [ ] Rename button size http://take.ms/3aCzV
- [ ] Nav input border color http://take.ms/M2Y45
- [ ] Filter dropdown thick border http://take.ms/yvB4V
- [ ] Modal close vertical alignment http://take.ms/LmtWQ
- [ ] http://dvlcore.dev/splash is super funky :)

### Feedback on the styleguide itself

- [ ] http://dvlcore.dev/headers is hard to parse. Maybe if we put each example in a well, or another kind of visual container?
- [ ] The hero unit header is kinda annoying to scroll through on every page. Should we consider only using it on the homepage?
- [ ] I'm impressed with what you've done with Erector and the helpers you've written here. Can I refactor some of them though? :)

kariannemah commented:
![img](http://take.ms/X9nCy)

The copy here is a bit confusing. In this particular example, does the minus icon indicate that more than one response has been selected and Clay is assigned to some of the them but not others?

ajb commented:
Some other thoughts... 

- The navbar went from 52px to 64px tall. (when rendered at desktop viewports) While I'm :+1: on making our font sizes bigger, the fact that we're using fixed navbars in a lot of places makes me want to keep it a bit smaller, in order to give ourselves more viewport space to work with. If you look at Screendoor's formbuilder when rendered with the new dvl-core, the navbar + bottom_status_bar probably take up 150px of vertical space in total, which negatively impacts the usability of the formbuilder, IMO.

- What about flashes? (https://github.com/dobtco/dvl-core/pull/83)

jrubenoff commented:
Thanks, Kari and Adam! Addressed most of your comments. Some of these problems occurred because I didn't specify Source Sans Pro correctly, so I'd recommend you take another look at headers, etc.

> There's a perceptible delay when checking radios and checkboxes

Can't reproduce this in Safari, FF, or Chrome. What browser are you using?

> Rename button size http://take.ms/3aCzV

It's supposed to be that large, to match the `h2`.

> Filter dropdown thick border

We can change to this if it's really an issue, but IMO it's uglier:

![screen shot 2015-06-30 at 4 06 43 pm](https://cloud.githubusercontent.com/assets/1328849/8444108/1c125304-1f42-11e5-8490-0cd0523a3219.png)

---

As for the splash footer, I wanted to take that out of `dvl-core`, but let me redesign it instead.

Let's create two variants of the hero image. Not sure how to change the layout depending on each page... can you handle that?

Feel free to refactor the partials... as long as the design doesn't change, I don't care.

Also, I barely touched flashes. Didn't want to step on the work you've done. We can merge that in after this.

ajb commented:
> As for the splash footer, I wanted to take that out of dvl-core, but let me redesign it instead.

We can take it out, but if we do, we should put it in a repository somewhere where all of the apps using it can vendor it in. (Splash, knowledge base.... maybe that's it.)

ajb commented:
> Can't reproduce this in Safari, FF, or Chrome. What browser are you using?

Chrome. This "issue" exists in `master` too, but I think I'm noticing this here because the element is bigger. The [transition](https://github.com/dobtco/dvl-core/blob/3bbfa95159d626252c55cef06bb7ec5cf0fce0a9/vendor/assets/stylesheets/dvl/core/forms.scss#L282) on the checkbox and radio icons just makes them feel sluggish. Maybe we can speed them up a little bit? I don't know, there has to be a reason why they feel more sluggish in this branch....

---

> It's supposed to be that large, to match the h2.

Right, I think it's too prominent for a button that will _maybe_ be clicked once. Can we just remove the border maybe?

---

> We can change to this if it's really an issue, but IMO it's uglier:

Unfortunately I think it is an issue... especially when combined with the "multiple sections" (visible in the assignment dropdown w/ teams) it looks really off. 

----


> Let's create two variants of the hero image. Not sure how to change the layout depending on each page... can you handle that?

Yup, of course.

---

> I'm not so sure about orange as a primary button color. Am I the only one that feels this way? 

You checked this off, but I don't see any code changes, or any response to this in your comments. 

---

![img](http://take.ms/tyFN6)

---

I finally grokked the thing about `.subtle` buttons being hidden instead of disabled... I still don't totally get why that behavior exists, or when we'd use it. But maybe that's just me.

---

![img](http://take.ms/6Egt4)

ajb commented:
Actually, the radio / checkbox behavior appears to be a lot better in Screendoor than in the style guide... maybe we should investigate during our check-in today.

aviv commented:
This looks great! Looking forward to having more re-usable components and design patterns/guidance.

jrubenoff commented:
I'm up for changing the filter dropdown, but since I don't have a problem with it, I wanted to make sure we're talking about the latest commit:

![filter_dropdown](https://cloud.githubusercontent.com/assets/1328849/8467667/3c59d3ba-2016-11e5-9008-e2f804e14420.png)

Is that still too weird? If so, no worries. I'll change it.

Just fixed everything else you mentioned.

ajb commented:
No; that solves my only real problem with it. I think I can get used to the thick border.

It looks like pretty much all of the comments have been addressed, besides the ones about the styleguide itself. I can work on those today.

ajb commented on vendor/assets/stylesheets/dvl/core.scss:
```diff
@@ -6,6 +6,7 @@
 @import "core/includes";
 @import "core/resets";
 @import "core/utils";
+@import "core/icons";
```
Let's leave these out of this manifest.

ajb commented on vendor/assets/stylesheets/dvl/core/icons.scss:
```diff
@@ -0,0 +1,179 @@
+@font-face {
+  font-family: 'dobt';
```
@jrubenoff: what do you think about using a data-URI to avoid more HTTP requests for this icon font? can Icomoon generate that for us?

jrubenoff replied:
Just saw this. Icomoon can't, but FontSquirrel can. I'll investigate.

ajb commented:
As you mentioned yesterday, I think we should add some documentation around icons. I'm not really sure how we're using font-awesome _and_ our own iconset at the same time.

jrubenoff commented:
Currently we're not using them at all, I just committed the fonts and styles to the repo as a starting point.

How about we delete the icons, merge this, and then add them back in a subsequent PR?

ajb commented:
That would certainly make sense :D

ajb commented:
Just made some commits where I said I would. Two more unimportant comments:

![img](http://take.ms/Yr4Qu)

---

![img](http://take.ms/V31LO)

ajb commented on spec/dummy/app/views/home/color.rb:
```diff
@@ -0,0 +1,104 @@
+
+  def uiColors
+    {
+      '$primaryColor' => '#697182',
```
It kinda sucks that we're duplicating these, huh? Let's at least add a code comment to alert maintainers that they need to be kept in-sync...

ajb commented on vendor/assets/stylesheets/dvl/core/lists.scss:
```diff
@@ -1,4 +1,40 @@
+  li {
+    position: relative;
+    counter-increment: list;
+    margin: 0 0 $lineHeight $rhythm * 5;
```
This is intentionally larger than for `formatted_list_ul`?

jrubenoff replied:
Yup. The dots in ULs are always a uniform width. The numbers in OLs can
increase to triple digits.

On Thu, Jul 2, 2015 at 10:54 AM, Adam Becker <notifications@github.com>
wrote:

> In vendor/assets/stylesheets/dvl/core/lists.scss
> <https://github.com/dobtco/dvl-core/pull/105#discussion_r33804254>:
>
> > +      right: 100%;
> > +      content: '';
> > +      width: $rhythm;
> > +      height: $rhythm;
> > +      border: 1px solid $darkestGray;
> > +      border-radius: 50%;
> > +    }
> > +  }
> > +}
> > +
> > +@mixin formatted_list_ol {
> > +  counter-reset: list;
> > +  li {
> > +    position: relative;
> > +    counter-increment: list;
> > +    margin: 0 0 $lineHeight $rhythm * 5;
>
> This is intentionally larger than for formatted_list_ul?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105/files#r33804254>.
>

ajb replied:
Really? What's the chance of us having a <ol> with more than 99 items in it?

On Thu, Jul 2, 2015 at 6:09 PM, Josh Rubenoff <notifications@github.com>
wrote:

> In vendor/assets/stylesheets/dvl/core/lists.scss
> <https://github.com/dobtco/dvl-core/pull/105#discussion_r33827812>:
>
> > +      right: 100%;
> > +      content: '';
> > +      width: $rhythm;
> > +      height: $rhythm;
> > +      border: 1px solid $darkestGray;
> > +      border-radius: 50%;
> > +    }
> > +  }
> > +}
> > +
> > +@mixin formatted_list_ol {
> > +  counter-reset: list;
> > +  li {
> > +    position: relative;
> > +    counter-increment: list;
> > +    margin: 0 0 $lineHeight $rhythm * 5;
>
> Yup. The dots in ULs are always a uniform width. The numbers in OLs can
> increase to triple digits.
> … <#14e50d12274262b0_>
> On Thu, Jul 2, 2015 at 10:54 AM, Adam Becker <notifications@github.com>
> wrote: In vendor/assets/stylesheets/dvl/core/lists.scss <
> https://github.com/dobtco/dvl-core/pull/105#discussion_r33804254>: > +
> right: 100%; > + content: ''; > + width: $rhythm; > + height: $rhythm; > +
> border: 1px solid $darkestGray; > + border-radius: 50%; > + } > + } > +} >
> + > +@mixin <https://github.com/mixin> formatted_list_ol { > +
> counter-reset: list; > + li { > + position: relative; > +
> counter-increment: list; > + margin: 0 0 $lineHeight $rhythm * 5; This is
> intentionally larger than for formatted_list_ul? — Reply to this email
> directly or view it on GitHub <
> https://github.com/dobtco/dvl-core/pull/105/files#r33804254>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105/files#r33827812>.
>



-- 
Adam Becker
(951) 9-BECKER
@AdamJacobBecker

jrubenoff replied:
Not us, but within user-generated content (project descriptions, etc…)

On Thu, Jul 2, 2015 at 3:10 PM, Adam Becker <notifications@github.com>
wrote:

> In vendor/assets/stylesheets/dvl/core/lists.scss
> <https://github.com/dobtco/dvl-core/pull/105#discussion_r33827910>:
>
> > +      right: 100%;
> > +      content: '';
> > +      width: $rhythm;
> > +      height: $rhythm;
> > +      border: 1px solid $darkestGray;
> > +      border-radius: 50%;
> > +    }
> > +  }
> > +}
> > +
> > +@mixin formatted_list_ol {
> > +  counter-reset: list;
> > +  li {
> > +    position: relative;
> > +    counter-increment: list;
> > +    margin: 0 0 $lineHeight $rhythm * 5;
>
> Really? What's the chance of us having a <ol> with more than 99 items in
> it?
> … <#14e50d2442fecc70_>
> On Thu, Jul 2, 2015 at 6:09 PM, Josh Rubenoff <notifications@github.com>
> wrote: In vendor/assets/stylesheets/dvl/core/lists.scss <
> https://github.com/dobtco/dvl-core/pull/105#discussion_r33827812>: > +
> right: 100%; > + content: ''; > + width: $rhythm; > + height: $rhythm; > +
> border: 1px solid $darkestGray; > + border-radius: 50%; > + } > + } > +} >
> + > +@mixin <https://github.com/mixin> formatted_list_ol { > +
> counter-reset: list; > + li { > + position: relative; > +
> counter-increment: list; > + margin: 0 0 $lineHeight $rhythm * 5; Yup. The
> dots in ULs are always a uniform width. The numbers in OLs can increase to
> triple digits. … <#14e50d12274262b0_> On Thu, Jul 2, 2015 at 10:54 AM, Adam
> Becker <notifications@github.com> wrote: In
> vendor/assets/stylesheets/dvl/core/lists.scss <
> https://github.com/dobtco/dvl-core/pull/105#discussion_r33804254>: > +
> right: 100%; > + content: ''; > + width: $rhythm; > + height: $rhythm; > +
> border: 1px solid $darkestGray; > + border-radius: 50%; > + } > + } > +} >
> + > +@mixin <https://github.com/mixin> <https://github.com/mixin>
> formatted_list_ol { > + counter-reset: list; > + li { > + position:
> relative; > + counter-increment: list; > + margin: 0 0 $lineHeight $rhythm
> * 5; This is intentionally larger than for formatted_list_ul? — Reply to
> this email directly or view it on GitHub <
> https://github.com/dobtco/dvl-core/pull/105/files#r33804254>. — Reply to
> this email directly or view it on GitHub <
> https://github.com/dobtco/dvl-core/pull/105/files#r33827812>.
>  -- Adam Becker (951) 9-BECKER @AdamJacobBecker
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105/files#r33827910>.
>

ajb replied:
I can't really see that ever being the case, tbh. Even the worst government
documents I've seen haven't done into the triple digits in a numbered list.

On Thu, Jul 2, 2015 at 6:11 PM, Josh Rubenoff <notifications@github.com>
wrote:

> In vendor/assets/stylesheets/dvl/core/lists.scss
> <https://github.com/dobtco/dvl-core/pull/105#discussion_r33827977>:
>
> > +      right: 100%;
> > +      content: '';
> > +      width: $rhythm;
> > +      height: $rhythm;
> > +      border: 1px solid $darkestGray;
> > +      border-radius: 50%;
> > +    }
> > +  }
> > +}
> > +
> > +@mixin formatted_list_ol {
> > +  counter-reset: list;
> > +  li {
> > +    position: relative;
> > +    counter-increment: list;
> > +    margin: 0 0 $lineHeight $rhythm * 5;
>
> Not us, but within user-generated content (project descriptions, etc…)
> … <#14e50d33d32431b5_>
> On Thu, Jul 2, 2015 at 3:10 PM, Adam Becker <notifications@github.com>
> wrote: In vendor/assets/stylesheets/dvl/core/lists.scss <
> https://github.com/dobtco/dvl-core/pull/105#discussion_r33827910>: > +
> right: 100%; > + content: ''; > + width: $rhythm; > + height: $rhythm; > +
> border: 1px solid $darkestGray; > + border-radius: 50%; > + } > + } > +} >
> + > +@mixin <https://github.com/mixin> formatted_list_ol { > +
> counter-reset: list; > + li { > + position: relative; > +
> counter-increment: list; > + margin: 0 0 $lineHeight $rhythm * 5; Really?
> What's the chance of us having a <ol> with more than 99 items in it? …
> <#14e50d2442fecc70_> On Thu, Jul 2, 2015 at 6:09 PM, Josh Rubenoff <
> notifications@github.com> wrote: In
> vendor/assets/stylesheets/dvl/core/lists.scss <
> https://github.com/dobtco/dvl-core/pull/105#discussion_r33827812>: > +
> right: 100%; > + content: ''; > + width: $rhythm; > + height: $rhythm; > +
> border: 1px solid $darkestGray; > + border-radius: 50%; > + } > + } > +} >
> + > +@mixin <https://github.com/mixin> <https://github.com/mixin>
> formatted_list_ol { > + counter-reset: list; > + li { > + position:
> relative; > + counter-increment: list; > + margin: 0 0 $lineHeight $rhythm
> * 5; Yup. The dots in ULs are always a uniform width. The numbers in OLs
> can increase to triple digits. … <#14e50d12274262b0_> On Thu, Jul 2, 2015
> at 10:54 AM, Adam Becker <notifications@github.com> wrote: In
> vendor/assets/stylesheets/dvl/core/lists.scss <
> https://github.com/dobtco/dvl-core/pull/105#discussion_r33804254>: > +
> right: 100%; > + content: ''; > + width: $rhythm; > + height: $rhythm; > +
> border: 1px solid $darkestGray; > + border-radius: 50%; > + } > + } > +} >
> + > +@mixin <https://github.com/mixin> <https://github.com/mixin> <
> https://github.com/mixin> formatted_list_ol { > + counter-reset: list; >
> + li { > + position: relative; > + counter-increment: list; > + margin: 0 0
> $lineHeight $rhythm * 5; This is intentionally larger than for
> formatted_list_ul? — Reply to this email directly or view it on GitHub <
> https://github.com/dobtco/dvl-core/pull/105/files#r33804254>. — Reply to
> this email directly or view it on GitHub <
> https://github.com/dobtco/dvl-core/pull/105/files#r33827812>. -- Adam
> Becker (951) 9-BECKER @AdamJacobBecker — Reply to this email directly or
> view it on GitHub <
> https://github.com/dobtco/dvl-core/pull/105/files#r33827910>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105/files#r33827977>.
>



-- 
Adam Becker
(951) 9-BECKER
@AdamJacobBecker

jrubenoff replied:
OK, just made it a bit smaller so it only accommodates double-digit numbers.

ajb commented on vendor/assets/stylesheets/dvl/core/lists.scss:
```diff
@@ -6,25 +42,20 @@ ul, ol {
-    list-style-type: disc;
+    @include formatted_list_ul;
   }
   ol {
```
Are we losing support for nested `ol`s?

jrubenoff replied:
I could add them back in, but do we ever use them?

On Thu, Jul 2, 2015 at 10:54 AM, Adam Becker <notifications@github.com>
wrote:

> In vendor/assets/stylesheets/dvl/core/lists.scss
> <https://github.com/dobtco/dvl-core/pull/105#discussion_r33804311>:
>
> >    ul {
> > -    list-style-type: disc;
> > +    @include formatted_list_ul;
> >    }
> >    ol {
>
> Are we losing support for nested ols?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105/files#r33804311>.
>

ajb replied:
No, I think it's fine to leave them out. Just was checking for clarification.

ajb commented:
Are `cssgradients` the only thing we're using modernizr for? If so, I'd like to replace our build with this one: http://modernizr.com/download/#-cssgradients-cssclasses

jrubenoff commented:
Sure, I just need those CSS classes.

On Thu, Jul 2, 2015 at 11:28 AM, Adam Becker <notifications@github.com>
wrote:

> Are cssgradients the only thing we're using modernizr for? If so, I'd
> like to replace our build with this one:
> http://modernizr.com/download/#-cssgradients-cssclasses
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105#issuecomment-118117027>.
>

jrubenoff commented:
I think we can merge this in whenever Screendoor's PR is ready.

ajb commented:
I think the content of this PR is ready, but I'd like a chance to quickly refactor a couple of things before merging:

- [ ] Tweak modernizr build
- [ ] Remove custom iconset

Can you also address this one item from above?

- [ ] [nav dropdown placement](https://camo.githubusercontent.com/e92ed7f2bc460fe5dcc079f90a08d11dcb264353/687474703a2f2f74616b652e6d732f5972345175)

ajb commented:
Another question: Adding `.modal_footer_actions` touches a lot of code, and I'm not entirely sure why we need it: https://github.com/dobtco/dvl-core/blob/redesign/vendor/assets/stylesheets/dvl/components/modals.scss#L145-L166

The comment on line 145 says "Footer (for actions)". And what's the different between footer_actions and footer_link?

dobthound commented on spec/dummy/lib/sass_variable_evaluator.rb:
```diff
@@ -0,0 +1,12 @@
+class SassVariableEvaluator < Sass::Tree::Visitors::Base
+  def visit_comment(node)
```
Unused method argument - `node`. If it's necessary, use `_` or `_node` as an argument name to indicate that it won't be used. You can also write as `visit_comment(*)` if you want the method to accept any arguments but don't care about them.

ajb commented:
Here are some bugs in IE9 (sorry):

- [ ] Dropdown carets http://take.ms/BZgsR
- [ ] checkbox checks http://take.ms/WR3qn
- [ ] icon vert align http://take.ms/dx0oH
- [ ] really grainy logos http://take.ms/eSTzs
- [ ] more icon alignment http://take.ms/P7qio
- [ ] even more icon alignment http://take.ms/5L2tF

Is there a way that we're vertically aligning things that doesn't work in these older browsers? I'd like to know, too...

ajb commented:
Just caught a code issue:

We appear to be distinguishing between our two pagination components using `+ .pagination_status` like so:

![img](http://take.ms/3HjTb)

This strikes me as not very maintainable and not very performant. Why not just create two separate components?

dobthound commented on vendor/assets/javascripts/dvl/plugins/modernizr_custom.js:
```diff
@@ -0,0 +1,4 @@
+/* Modernizr 2.8.3 (Custom Build) | MIT & BSD
+ * Build: http://modernizr.com/download/#-cssgradients-cssclasses
+ */
+;window.Modernizr=function(a,b,c){function v(a){j.cssText=a}function w(a,b){return v(m.join(a+";")+(b||""))}function x(a,b){return typeof a===b}function y(a,b){return!!~(""+a).indexOf(b)}function z(a,b,d){for(var e in a){var f=b[a[e]];if(f!==c)return d===!1?a[e]:x(f,"function")?f.bind(d||b):f}return!1}var d="2.8.3",e={},f=!0,g=b.documentElement,h="modernizr",i=b.createElement(h),j=i.style,k,l={}. ...
```
Line is too long.<br>Unnecessary semicolon.<br>Missing semicolon.<br>Unexpected '~'.<br>Expected '{' and instead saw 'return'.<br>Confusing use of '!'.<br>The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.<br>Expected '{' and instead saw 'throw'.<br>Missing '()' invoking a constructor.<br>A constructor name should start with an uppercase letter.<br>Expected an assignment or function call and instead saw an expression.<br>Expected '{' and instead saw 'u'.<br>Expected '{' and instead saw 'for'.<br>Wrap an immediate function invocation in parens to assist the reader in understanding that the expression is the result of a function, and not the function itself.<br>'w' is defined but never used.<br>'z' is defined but never used.<br>'l' is defined but never used.<br>'o' is defined but never used.<br>'p' is defined but never used.

jrubenoff commented:
I just added vertical alignment mixins for `margin-top` and `top`, to fix IE9 issues.

I was waiting to refactor `.pagination_status` in a future PR to minimize code changes in Screendoor, but since we apparently only use it once, I just refactored it. Also renamed the modal footer containers to `.modal_footer_primary` and `.modal_footer_secondary`... they're for primary and secondary actions.

You can add `-ms-interpolation-mode: bicubic` for images, which I just did, but it still looks pretty bad. To improve it further, I think we would need to use IE filters, which require extra inline styles, so I'd rather not.

That's everything.

ajb commented:
It works in IE9! Consider me extremely impressed :)

Here's one bug though:

![img](http://take.ms/ot21R)

jrubenoff commented:
Just fixed that.

ajb commented on vendor/assets/stylesheets/dvl/core/grid.scss:
```diff
@@ -11,6 +11,9 @@
   padding: 0;
   list-style: none; /* 2 */
   font-size: 0; /* 3 */
+  @media screen and (min-width: $lapWidth) {
```
I'm not sure I follow. Isn't the point of `rem` units that they're already relative to screen size?

I find this behavior to be confusing and adding unnecessary complexity. Is it necessary to have?

jrubenoff replied:
Nope, units are relative to font size, so they don't change when the viewport changes.

It's pretty simple. Larger screen sizes = larger gutter width.

ajb replied:
Are we no longer adjusting our base font size based on viewport? Because if so, then that's a whole other issue...

ajb commented on vendor/assets/stylesheets/dvl/core/typography.scss:
```diff
@@ -6,18 +6,19 @@
 \* ================================================================== */
 
-html { font-size: 75%; } // 12px
-@media only screen and (min-width: 320px) { html { font-size: $maxFontSize * 0.8125; } } // 13px
```
@jrubenoff: I am only realizing now that this has been removed. Can you explain?

ajb replied:
I'm pretty sure this is what's causing bad layouts in small viewports: https://github.com/dobtco/screendoor-v2/pull/2153#issuecomment-117712592

jrubenoff replied:
The bugs you mentioned in that comment have been fixed.

On Wed, Jul 8, 2015 at 11:15 AM, Adam Becker <notifications@github.com>
wrote:

> In vendor/assets/stylesheets/dvl/core/typography.scss
> <https://github.com/dobtco/dvl-core/pull/105#discussion_r34180043>:
>
> > @@ -6,18 +6,19 @@
> >     http://modularscale.com/scale/?px1=12&px2=20&ra1=1.333&ra2=0
> >  \* ================================================================== */
> >
> > -html { font-size: 75%; } // 12px
> > -@media only screen and (min-width: 320px) { html { font-size: $maxFontSize * 0.8125; } } // 13px
>
> I'm pretty sure this is what's causing bad layouts in small viewports: dobtco/screendoor-v2#2153
> (comment)
> <https://github.com/dobtco/screendoor-v2/pull/2153#issuecomment-117712592>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dobtco/dvl-core/pull/105/files#r34180043>.
>

ajb replied:
Alright, after a bit of research, I'm fine with this, as long as we test our mobile layouts thoroughly. 

Although with the way it is now, is there any functional difference between `rem`s and `px`s? If we need to stick with `rem`s because switching to pixels would be too big of a pain in the butt, that's fine too.

jrubenoff replied:
`rem`s are more flexible, especially if we decide to change the root font size.

ajb commented:
I know Screendoor's PR has a couple of outstanding issues, but it seems like this one is good to merge.

ajb commented:
- [ ] selects in filter forms:

![img](http://take.ms/faoS7)

I'm not sure why there is a margin being applied, so I'll wait to make any changes.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+  var Tab = function (element) {
+    // jscs:disable requireDollarBeforejQueryAssignment
+    this.element = $(element)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    // jscs:disable requireDollarBeforejQueryAssignment
+    this.element = $(element)
+    // jscs:enable requireDollarBeforejQueryAssignment
+  }
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    // jscs:enable requireDollarBeforejQueryAssignment
+  }
+
+  Tab.VERSION = '3.3.5'
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+  Tab.VERSION = '3.3.5'
+
+  Tab.TRANSITION_DURATION = 150
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+  Tab.TRANSITION_DURATION = 150
+
+  Tab.prototype.show = function () {
+    var $this    = this.element
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+  Tab.prototype.show = function () {
+    var $this    = this.element
+    var $ul      = $this.closest('ul:not(.dropdown_menu)')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+  Tab.prototype.show = function () {
+    var $this    = this.element
+    var $ul      = $this.closest('ul:not(.dropdown_menu)')
+    var selector = $this.data('target')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+    if (!selector) {
+      selector = $this.attr('href')
+      selector = selector && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
```
Line is too long.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    var selector = $this.data('target')
+
+    if (!selector) {
+      selector = $this.attr('href')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      selector = selector && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
+    }
+
+    if ($this.parent('li').hasClass('active')) return
```
Expected '{' and instead saw 'return'.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+    if ($this.parent('li').hasClass('active')) return
+
+    var $previous = $ul.find('.active:last a')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    var $previous = $ul.find('.active:last a')
+    var hideEvent = $.Event('hide.bs.tab', {
+      relatedTarget: $this[0]
+    })
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    })
+    var showEvent = $.Event('show.bs.tab', {
+      relatedTarget: $previous[0]
+    })
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      relatedTarget: $previous[0]
+    })
+
+    $previous.trigger(hideEvent)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    })
+
+    $previous.trigger(hideEvent)
+    $this.trigger(showEvent)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    $previous.trigger(hideEvent)
+    $this.trigger(showEvent)
+
+    if (showEvent.isDefaultPrevented() || hideEvent.isDefaultPrevented()) return
```
Expected '{' and instead saw 'return'.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+    if (showEvent.isDefaultPrevented() || hideEvent.isDefaultPrevented()) return
+
+    var $target = $(selector)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+    var $target = $(selector)
+
+    this.activate($this.closest('li'), $ul)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      $previous.trigger({
+        type: 'hidden.bs.tab',
+        relatedTarget: $this[0]
+      })
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      $this.trigger({
+        type: 'shown.bs.tab',
+        relatedTarget: $previous[0]
+      })
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+        type: 'shown.bs.tab',
+        relatedTarget: $previous[0]
+      })
+    })
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+        relatedTarget: $previous[0]
+      })
+    })
+  }
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+  }
+
+  Tab.prototype.activate = function (element, container, callback) {
+    var $active    = container.find('> .active')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+  Tab.prototype.activate = function (element, container, callback) {
+    var $active    = container.find('> .active')
+    var transition = callback
+      && $.support.transition
```
Bad line breaking before '&&'.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+    var $active    = container.find('> .active')
+    var transition = callback
+      && $.support.transition
+      && ($active.length && $active.hasClass('fade') || !!container.find('> .fade').length)
```
Line is too long.<br>Bad line breaking before '&&'.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+          .removeClass('active')
+        .end()
+        .find('[data-toggle="tab"]')
+          .attr('aria-expanded', false)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      element
+        .addClass('active')
+        .find('[data-toggle="tab"]')
+          .attr('aria-expanded', true)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+          .attr('aria-expanded', true)
+
+      if (transition) {
+        element[0].offsetWidth // reflow for transition
```
Expected an assignment or function call and instead saw an expression.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+      if (transition) {
+        element[0].offsetWidth // reflow for transition
+        element.addClass('in')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+        element[0].offsetWidth // reflow for transition
+        element.addClass('in')
+      } else {
+        element.removeClass('fade')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+            .addClass('active')
+          .end()
+          .find('[data-toggle="tab"]')
+            .attr('aria-expanded', true)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+            .attr('aria-expanded', true)
+      }
+
+      callback && callback()
```
Expected an assignment or function call and instead saw an expression.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      $active
+        .one('bsTransitionEnd', next)
+        .emulateTransitionEnd(Tab.TRANSITION_DURATION) :
+      next()
```
Expected an assignment or function call and instead saw an expression.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+        .emulateTransitionEnd(Tab.TRANSITION_DURATION) :
+      next()
+
+    $active.removeClass('in')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      next()
+
+    $active.removeClass('in')
+  }
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+  function Plugin(option) {
+    return this.each(function () {
+      var $this = $(this)
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+  function Plugin(option) {
+    return this.each(function () {
+      var $this = $(this)
+      var data  = $this.data('bs.tab')
```
Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      var $this = $(this)
+      var data  = $this.data('bs.tab')
+
+      if (!data) $this.data('bs.tab', (data = new Tab(this)))
```
Expected '{' and instead saw '$this'.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+      var data  = $this.data('bs.tab')
+
+      if (!data) $this.data('bs.tab', (data = new Tab(this)))
+      if (typeof option == 'string') data[option]()
```
Expected '{' and instead saw 'data'.<br>Missing semicolon.

dobthound commented on vendor/assets/javascripts/dvl/components/tabs.js:
```diff
@@ -0,0 +1,155 @@
+
+      if (!data) $this.data('bs.tab', (data = new Tab(this)))
+      if (typeof option == 'string') data[option]()
+    })
```
Missing semicolon.<br>Too many errors. (80% scanned).

ajb commented:
Merged in #114.
